src/object.d

```diff
@@ -155,15 +164,269 @@ auto opEquals(Object lhs, Object rhs)
     return lhs.opEquals(rhs) && rhs.opEquals(lhs);
```

Comment:
Hmm. Good catch. There probably should be a check that the type has `opEquals`. Certainly, we're not trying to use UFCS for anything here, and having it kick in would be problematic.